iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -249,6 +228,34 @@ private async void ConnectionStatusChangeHandler(ConnectionStatus status, Connec
             }
         }
 
+        private async Task GetTwinOnConnectAsync(CancellationToken cancellationToken)
+        {

```

Comment:
nit: for async methods being called from an async void callback, it is generally safer to add a lock so that we're not executing them multiple times for a single event. Eg. if multiple coe paths result in the connection status callback being triggered, we don't want each of them to be executing this block. For our specific case, ideally, the SDK shouldn't be firing multiple events for the same event change, but unless we have that explicitly tested and documented, I think it is worth handling that scenario.